TrackletAlgorithm/TrackMerger.cc

```diff
@@ -0,0 +1,49 @@
+#include "TrackMerger.h"
+
+void TrackMerger(const BXType bx, TrackFit::TrackWord trackWord [kMaxProc], TrackFit::BarrelStubWord barrelStubWords[4][kMaxProc], 
+TrackFit::DiskStubWord diskStubWords[4][kMaxProc]){
```

Comment:
You will need two sets of the arrays you have defined here, one for the input tracks and one for the output tracks. So I think the interface of this function will look something like the following:
```c++
void TrackMerger(
  const BXType bx,
  const TrackFit::TrackWord trackWord [kMaxProc],
  const TrackFit::BarrelStubWord barrelStubWords[4][kMaxProc],
  const TrackFit::DiskStubWord diskStubWords[4][kMaxProc],
  BXType &bx_o,
  TrackFit::TrackWord trackWord_o [kMaxProc],
  TrackFit::BarrelStubWord barrelStubWords_o[4][kMaxProc],
  TrackFit::DiskStubWord diskStubWords_o[4][kMaxProc]
){
…
```

This example also includes the `bx_o` port that all processing modules should have. It should be assigned the value of `bx` at the end of processing, e.g.:
https://github.com/cms-L1TK/firmware-hls/blob/3ebd273bdff13cf027166db804c7ca1fc38791a1/TrackletAlgorithm/TrackBuilder.h#L309